### PR TITLE
Experience Leaderboard

### PR DESCRIPTION
**Pull request by DynxstyGIT**
image

### CODE REVIEW

File: src/main/java/net/javadiscord/javabot/systems/help/dao/HelpAccountRepository.java [hunk 1]
```diff
@@ -66,6 +68,40 @@ public Optional<HelpAccount> getByUserId(long userId) throws SQLException {
...
 	}
 
+	/**
+	 * Gets as many Accounts as specified.
+	 *
+	 * @param page    The page.
+	 * @param size    The amount of account to return.
+	 * @return A {@link List} containing the specified amount of {@link HelpAccount}s.
+	 * @throws SQLException If an error occurs.
+	 */
```
Comment (MoonTM-GIT) [inline src/main/java/net/javadiscord/javabot/systems/help/dao/HelpAccountRepository.java:78]: I think this sounds a bit better and is also more similar to other Javadoc in this class
```suggestion
	/**
	 * Gets a specified amount of {@link HelpAccount}s.
	 *
	 * @param page    The page.
	 * @param size    The amount of {@link HelpAccount}s to return.
	 * @return A {@link List} containing the specified amount of {@link HelpAccount}s.
	 * @throws SQLException If an error occurs.
	 */
```

**Review by MoonTM-GIT — CHANGES_REQUESTED** [review]

File: src/main/java/net/javadiscord/javabot/systems/help/commands/subcommands/ExperienceLeaderboardSubcommand.java [hunk 2]
```diff
@@ -63,17 +64,18 @@ public static void handleButtons(ButtonInteractionEvent event, String[] id) {
...
-		List<HelpAccount> accounts = dao.getAccountsWithRank(Math.min(page, maxPage), PAGE_SIZE);
+		int maxPage = dao.getTotalAccounts() / PAGE_SIZE;
+		List<HelpAccount> accounts = dao.getAccounts(Math.min(page, maxPage), PAGE_SIZE);
 		EmbedBuilder builder = new EmbedBuilder()
 				.setTitle("Experience Leaderboard")
 				.setColor(Bot.config.get(guild).getSlashCommand().getDefaultColor())
 				.setFooter(String.format("Page %s/%s", Math.min(page, maxPage), maxPage));
 		accounts.forEach(account -> {
 			Map.Entry<Long, Double> currentRole = account.getCurrentExperienceGoal(guild);
+			System.out.println(guild.getJDA().getUserCache().toString());
```
Comment (danthe1st) [inline src/main/java/net/javadiscord/javabot/systems/help/commands/subcommands/ExperienceLeaderboardSubcommand.java]: Don't print debug output to `System.out` in production. if that should be kept, use a logger.

**Review by danthe1st — CHANGES_REQUESTED** [review]

File: src/main/java/net/javadiscord/javabot/systems/help/commands/subcommands/ExperienceLeaderboardSubcommand.java (hunk 2 above)
Comment (DynxstyGIT) [reply src/main/java/net/javadiscord/javabot/systems/help/commands/subcommands/ExperienceLeaderboardSubcommand.java]: Oh, lmao. Forgot to remove that

**Comment by MoonTM-GIT** [comment]
test

**Comment by MoonTM-GIT** [comment]
test

File: src/main/java/net/javadiscord/javabot/systems/help/commands/subcommands/ExperienceLeaderboardSubcommand.java (hunk 2 above)
Comment (MoonTM-GIT) [reply src/main/java/net/javadiscord/javabot/systems/help/commands/subcommands/ExperienceLeaderboardSubcommand.java]: This was already removed in 11bf6bd7ef8a35f9a1fe4fdec7aadb593a690a2b

**Review by MoonTM-GIT — APPROVED** [review]

**Pull request merged**